src/Imgix/UrlBuilder.php

```diff
@@ -5,50 +5,34 @@
 class UrlBuilder {
 
     private $currentVersion = "2.3.0";
-    private $domains;
+    private $domain;
     private $useHttps;
     private $signKey;
-    private $shardStrategy;
 
-    private $shardCycleNextIndex = 0;
+    public function __construct($domain, $useHttps = false, $signKey = "", $includeLibraryParam = true) {
 
-    public function __construct($domains, $useHttps = false, $signKey = "", $shardStrategy = ShardStrategy::CRC, $includeLibraryParam = true) {
-        if (!is_array($domains)) {
-            $this->domains = array($domains);
-        } else {
-            $warning_message = "Warning: Domain sharding has been deprecated and will be removed in the next major version.";
-            @trigger_error($warning_message, E_USER_DEPRECATED);
-            $this->domains = $domains;
+        if (!is_string($domain)) {
+            throw new \InvalidArgumentException("UrlBuilder must be passed a valid string domain");
```

Comment:
*pushes up glasses*

Can we change this message to "UrlBuilder must be passed a string domain"? The validity of the string isn't technically relevant here, since validity isn't checked until L19.

Reply:
I did that to be consistent with [imgix-core-js](https://github.com/imgix/imgix-core-js/blob/87bb830b729ca0b3684f95ac470293729737fb7a/src/imgix-core-js.js#L41) but I see your point. If we change it here, should we carry it over everyone else that applies?

Reply:
Ah dinger, shoulda caught it there. I'd say change it here and make a note somewhere to roll that in the next time we change something in imgix-core-js. I don't think it's worth downshifting just to go back and change that, I'd rather you keep forging ahead.